janusgraph-solr/src/main/java/org/janusgraph/diskstorage/solr/SolrIndex.java

```diff
@@ -332,6 +335,8 @@ public void mutate(Map<String, Map<String, IndexMutation>> mutations, KeyInforma
                 String collectionName = stores.getKey();
                 String keyIdField = getKeyFieldId(collectionName);
 
+                Map<String, String> fieldDeletes = Collections.unmodifiableMap(new HashMap<String, String>(1) {{ put("set", null); }});
```

Comment:
Add `final`. Also I was suggesting `ImmutableMap.of(key,value)` to be more concise than `new HashMap...`, not because this should be immutable, so since ImmutableMap isn't possible can you revert back e.g. remove `Collections.unmodifiableMap` here and below? Sorry for the confusion, I should have been more clear in my comments.